api/src/main/java/org/openmrs/validator/ConditionValidator.java

```diff
@@ -44,7 +46,7 @@ public void validate(Object object, Errors errors) {
 			throw  new IllegalArgumentException("The object parameter should be of type " + Condition.class);
 		}
 		Condition condition = (Condition) object;
-		if(condition.getCondition() == null){
+		if(condition.getCondition() == null || (condition.getClinicalStatus().equals(ConditionClinicalStatus.INACTIVE) && condition.getOnsetDate().compareTo(condition.getEndDate()) > 0)){
```

Comment:
Don't do a OR statement here, do a separate validation under those that are there already.